app/src/main/java/com/mobileproto/hireddit/hireddit/reddit/HighestUpvoteCommentAsync.java

```diff
@@ -0,0 +1,109 @@
+package com.mobileproto.hireddit.hireddit.reddit;
+
+import android.os.AsyncTask;
+import android.util.Log;
+
+import org.jsoup.Jsoup;
+import org.jsoup.nodes.Document;
+import org.jsoup.nodes.Element;
+
+import java.io.IOException;
+import java.util.List;
+
+/**
+ * Created by david on 12/6/15.
+ */
+public class HighestUpvoteCommentAsync {
+
+    private static final String DEBUG_TAG = "Upvote Debug";
+    private static final String ERROR_TAG = "Upvote Error";
+
+    private RedditSearcher.CommentCallback myCommentCallback;
+    private List<String[]> allComments;
+    private Integer[] scores;
+
+    private class SingleCommentSearch extends AsyncTask<Void, Void, Integer> {
+        String url;
+        int index;
+
+        public SingleCommentSearch(String url, int index) {
+            this.url = url;
+            this.index = index;
+        }
+
+        @Override
+        protected Integer doInBackground(Void... params) {
+            int score = 0;
+            try {
+                Log.d(DEBUG_TAG, url);
+                Document doc = Jsoup.connect(url).get();
+                Element commentArea = doc.select("div.content").first().select("div.commentarea").first();
+                Element siteTable = commentArea.select("div.sitetable.nestedlisting").first();
+                Element tagline = siteTable.select("p.tagline").first();
+                Element scoreElement = tagline.select("span.score.unvoted").first();
```

Comment:
question: would you be saving space if you got the element by doing everything in one line so you're not storing a bunch of extra elements? (i think the answer is probably no but ill ask anyway)